packages/babel-plugin/src/styled/__tests__/call-expression.test.ts

```diff
@@ -538,4 +538,87 @@ describe('styled object call expression', () => {
 
     expect(actual).toIncludeMultiple(['color:black', 'font-weight:400', 'background-color:white']);
   });
+
+  it('should transform variable in a nested template literal', () => {
+    // this is the output of applying @atlaskit/tokens babel plugin
+    // (as of v1.0.0) to some code similar to this:
+    //     styled.div({
+    //         backgroundColor: token('some.token', color),
+    //     })
+
+    const actual = transform(`
+      import { styled } from '@compiled/react';
+
+      const color = 'red';
+
+      const Component = styled.div({
+        backgroundColor: \`var(--my-variable, \${color})\`,
+      });
+    `);
+
+    expect(actual).toInclude('{background-color:var(--my-variable,red)}');
+  });
+
+  it('should transform variable in a heavily nested template literal', () => {
+    // corresponds to
+    //     styled.div({
+    //         boxShadow: `0 8px ${token('some.token', color)}`
+    //     })
+
+    const actual = transform(`
+      import { styled } from '@compiled/react';
+
+      const color = 'red';
+
+      const Component = styled.div({
+        boxShadow: \`0 8px \${\`var(--my-variable, \${color})\`}\`,
+      });
+    `);
+
+    expect(actual).toInclude('{box-shadow:0 8px var(--my-variable,red)}');
+  });
+
+  it('should transform variables within nested template literals that are all in an interpolation function', () => {
+    // corresponds to
+    //     styled.div<{ isActive: boolean }>({
+    //         boxShadow: (props) =>
+    //             props.isActive
+    //                 ? `0 ${size}px ${token('some.token', color)}`
+    //                 : token('some.other.token', color2),
+    //     })
+
+    const actual = transform(`
+      import { styled } from '@compiled/react';
+
+      const color = 'red';
+      const color2 = 'blue';
+      const size = 5;
+
+      const Component = styled.div<{ isActive: boolean }>({
+        boxShadow: (props) =>
+            props.isActive
+                ? \`0 \${size}px \${\`var(--my-variable, \${color})\`}\`
+                : \`var(--my-other-variable, \${color2})\`,
+      });
+    `);
+
+    // We currently don't statically evaluate color, color2, or size here
+    expect(actual).toMatchInlineSnapshot(`
```

Comment:
Ah sorry yep, this is one of the cases where this PR doesn't have any effect. I added it to check that nothing was being silently deleted. Should I still keep test cases like these? 🤔

Reply:
If I understand it correctly, you want to make sure nothing gets accidentally deleted by `@compiled/babel-plugin`. The thing is `@compiled/babel-plugin` won't even recognise the syntax because the code has TypeScript `<{ isActive: boolean }>`. However,[ transform function](https://github.com/atlassian-labs/compiled/blob/master/packages/babel-plugin/src/test-utils.ts#L15) isn't configured to  parse TypeScript file so `@compiled/babel-plugin` will just bail out.

Reply:
you can configure the transform function to support TS:

```
const actual = transform(
  `
  import '@compiled/react';

  const styles = { color: 'red' } as const;

  <div css={{ ...styles }} />;
`,
  {
    parserBabelPlugins: ['typescript', 'jsx'],
  }
);
```